src/tape_drivers/linux/sg/sg_tape.c

```diff
@@ -118,31 +121,25 @@ static inline int _parse_logPage(const unsigned char *logdata,
 								 const uint16_t param, uint32_t *param_size,
 								 unsigned char *buf, const size_t bufsize)
 {
-	uint16_t page_len, param_code, param_len;
-	uint32_t i;
+	const uint16_t page_len = ((uint16_t)logdata[2] << 8) | (uint8_t)logdata[3];
+	uint16_t param_code, param_len;
+	uint32_t i = LOG_PAGE_HEADER_SIZE;
 	uint32_t ret = -EDEV_INTERNAL_ERROR;
 
-	page_len = ((uint16_t)logdata[2] << 8) + (uint16_t)logdata[3];
-	i = LOG_PAGE_HEADER_SIZE;
-
 	while(i < page_len)
 	{
-		param_code = ((uint16_t)logdata[i] << 8) + (uint16_t)logdata[i+1];
+		param_code = ((uint16_t)logdata[i] << 8) | (uint8_t)logdata[i+1];
 		param_len  = (uint16_t)logdata[i + LOG_PAGE_PARAMSIZE_OFFSET];
 
-		if(param_code == param)
+		if (param_code == param)
 		{
 			*param_size = param_len;
-			if(bufsize < param_len){
-				memcpy(buf, &logdata[i + LOG_PAGE_PARAM_OFFSET], bufsize);
-				ret = -EDEV_INTERNAL_ERROR;
-				break;
-			} else {
-				memcpy(buf, &logdata[i + LOG_PAGE_PARAM_OFFSET], param_len);
-				ret = DEVICE_GOOD;
-				break;
-			}
+			const size_t copy_len = (bufsize < param_len) ? bufsize : param_len;
```

Comment:
I cannot accept this block. Why you need to evaluate `(bufsize < param_len)` twice in this block ?

Please revert.